### PR TITLE
update mlego/m48 to data driven

### PR DESCRIPTION
**Pull request by alinelena**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/mlego/m48/config.h
Comment (tzarc) [inline keyboards/mlego/m48/config.h:1]: Delete this file.

**Review by tzarc — CHANGES_REQUESTED** [review]

File: keyboards/mlego/m48/config.h
Comment (alinelena) [reply keyboards/mlego/m48/config.h:1]: done thank you!

**Comment by fauxpark** [comment]
As of #20323 data-driven keyboard refactors should target `develop`.

File: keyboards/mlego/m48/rev1/info.json [hunk 1]
```diff
@@ -10,15 +10,111 @@
     "diode_direction": "COL2ROW",
     "encoder": {
         "rotary": [
-            {"pin_a": "A0", "pin_b": "B8"}
+            {
+                "pin_a": "A0",
+                "pin_b": "B8"
+            }
```
Comment (fauxpark) [inline keyboards/mlego/m48/rev1/info.json:16]:
```suggestion
            {"pin_a": "A0", "pin_b": "B8"}
```

`qmk format-json` should help here.

**Comment by alinelena** [comment]
I will reopen this when https://github.com/qmk/qmk_firmware/pull/20224 is merged

**Pull request closed without merging**